### PR TITLE
fix: 403 forbidden on device_type UNKNOWN, changed to ANDROID, fixes #115

### LINKED ISSUE #115: 403 on token refresh
Hello
I still have a 403 error with the last version after the PR for #112 and #113 

```
toogoodtogo-watcher_1  | Error during request:
toogoodtogo-watcher_1  | POST https://apptoogoodtogo.com/api/auth/v2/loginByEmail
toogoodtogo-watcher_1  | {
toogoodtogo-watcher_1  |     "device_type": "UNKNOWN",
toogoodtogo-watcher_1  |     "email": "*****",
toogoodtogo-watcher_1  |     "password": "*****"
toogoodtogo-watcher_1  | }
toogoodtogo-watcher_1  | 
toogoodtogo-watcher_1  | HTTPError: Response code 403 (Forbidden)
toogoodtogo-watcher_1  |     at Request.<anonymous> (/home/node/app/node_modules/got/dist/source/as-promise/index.js:117:42)
toogoodtogo-watcher_1  |     at processTicksAndRejections (internal/process/task_queues.js:95:5)

```

When I try to POST on Insomnia I obtain this response asking me to resolve a captcha

```
<html><head><title>apptoogoodtogo.com</title><style>#cmsg{animation: A 1.5s;}@keyframes A{0%{opacity:0;}99%{opacity:0;}100%{opacity:1;}}</style></head><body style="margin:0"><p id="cmsg">Please enable JS and disable any ad blocker</p><script>var dd={'cid':'AHrlqAAAAAMANoQuwp2NxI8ATZMnzg==','hsh':'1D42C2CA6131C526E09F294FE96F94','t':'fe','r':'b','s':35702,'host':'geo.captcha-delivery.com'}</script><script src="https://ct.captcha-delivery.com/c.js"></script></body></html>
```

Maybe we can re-open #114?

### PR DESCRIPTION
**Pull request by dennisvanderpool**
Credits go to @navimondo @andrzejpodgorny

### CODE REVIEW

**Comment by unixfox** [comment]
Sending ANDROID device while using a user agent from an iphone (https://github.com/iv-org/invidious/blob/master/src/invidious/yt_backend/youtube_api.cr#L410-L418) doesn't seem correct to me.

**Comment by unixfox** [comment]
Sending ANDROID device while using a user agent from an iphone (https://github.com/marklagendijk/node-toogoodtogo-watcher/blob/master/lib/api.js#L9) doesn't seem correct to me.

**Comment by dennisvanderpool** [comment]
@unixfox I did modify user agent in config.json as well, I did not include it in the PR though.

    "headers": {                                                                                                                                                                          "User-Agent": "TooGoodToGo/21.9.0 (813) (iPhone/iPhone 12 (GSM); iOS 15.1; Scale/2.00)"                                                                                           },

**Comment by andrzejpodgorny** [comment]
Yep, i would suggest using "IOS".

**Comment by dennisvanderpool** [comment]
@unixfox Can you try with the User-Agent I used?

**Comment by unixfox** [comment]
> @unixfox I did modify user agent in config.json as well, I did not include it in the PR though.
> 
> ```
> "headers": {                                                                                                                                                                          "User-Agent": "TooGoodToGo/21.9.0 (813) (iPhone/iPhone 12 (GSM); iOS 15.1; Scale/2.00)"                                                                                           },   
> ```

It makes more sense to send IOS or IPHONE if that's the correct device name to be used.

**Comment by dennisvanderpool** [comment]
Yeah, I agree. Sending device type ANDROID with an user agent of iOS is weird 😂

**Comment by dennisvanderpool** [comment]
Ok, I can update the PR with an adjusted config.defaults.json, but what user agent shall I paste in?

**Comment by dennisvanderpool** [comment]
Maybe @marklagendijk can join the discussion as well 😊

**Comment by andrzejpodgorny** [comment]
I would replace this line: https://github.com/marklagendijk/node-toogoodtogo-watcher/blob/3e6f6807ed6a75f0a2a888977a86d935e9d9982c/lib/api.js#L35

for `device_type: "credentials.device_type` and pass parameter in config.json file.

File: lib/api.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ function loginByEmail() {
   return api
     .post("auth/v2/loginByEmail", {
       json: {
-        device_type: "UNKNOWN",
+        device_type: "ANDROID",
```
Comment (jfadeuilhe) [inline lib/api.js:35]: Maybe : device_type: credentials.device_type, and put "device_type": "ANDROID", in config.json

**Comment by marklagendijk** [comment]
Thanks for the PR. 

Because it needed some improvements, I applied those directly on the code, instead of going through this PR.
* `IOS` is used as default, to match with the default user-agent header.
* It is made configurable.

**Pull request closed without merging**